Library/Homebrew/download_strategy.rb

```diff
@@ -269,6 +269,17 @@ def initialize(name, resource)
 
   def fetch
     ohai "Downloading #{@url}"
+
+    url = resolve_url(@url)
+    if url != @url
+      if @url.start_with?("https://") && url.start_with?("http://") &&
+         !ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil?
```

Comment:
I'd maybe tweak the indentation here so it's more obvious it's grouped with the previous line.